util/src/link/image.rs

```diff
@@ -0,0 +1,542 @@
+use std::{
+    fmt::{self, Display},
+    num::NonZeroU64,
+};
+
+use twilight_model::{
+    channel::message::{
+        sticker::{StickerBannerAssetId, StickerFormatType, StickerId},
+        Sticker,
+    },
+    guild::{Emoji, Guild, Member, Role},
+    id::{ApplicationId, EmojiId, GuildId, RoleId, UserId},
+    oauth::{team::Team, CurrentApplicationInfo},
+    user::User,
+};
+
+const BASE_URL: &'static str = "https://cdn.discordapp.com/";
+
+/// The extension used for non-animated images
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Extension {
+    /// `.png`
+    PNG,
+    /// `.jpg`
+    JPEG,
+    /// `.webp`
+    WEBP,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum _Extension {
```

Comment:
why do we need two extension enums?

Reply:
the _Extension has GIF variant as well, adding that to the public enum would be confusing because people aren’t supposed to pick GIF manually (might lead to invalid links), the GIF variant should only be picked if the endpoint actually supports GIF, which the api checks itself